src/lib/expression/evaluation/expression_functors.hpp

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include "expression_result.hpp"
+
+namespace opossum {
+
+template<typename T> constexpr bool is_logical_operand = std::is_same_v<int32_t, T> || std::is_same_v<NullValue, T>;
+
+bool to_bool(const bool value) { return value; }
+bool to_bool(const NullValue& value) { return false; }
```

Comment:
Wherever you use this, this logic might make sense, but as a general function, I don't think this is correct. `is_true` could be a better name.